include/zephyr/pm/policy.h

```diff
@@ -125,6 +125,9 @@ void pm_policy_state_lock_get(enum pm_state state, uint8_t substate_id);
  */
 void pm_policy_state_lock_put(enum pm_state state, uint8_t substate_id);
 
+void pm_policy_state_lock_get_list(struct pm_state_constraint_list* constraints);
+void pm_policy_state_lock_put_list(struct pm_state_constraint_list* constraints);
```

Comment:
Since we are not actually passing a "list" of states `(enum pm_state *states, uint8_t *substate_ids, uint8_t size)` we could instead name the function according to the type passed to it:
```
void pm_policy_state_constraint_list_lock_get()
void pm_policy_state_constraint_list_lock_get()
```
and also, the name `struct pm_state_constraint_list` could be `struct pm_state_constraints`, which I think looks nicer, and sounds better if used in the new APIs
```
void pm_policy_state_constraints_lock_get()
```
and even remove the `lock` part:
```
void pm_policy_state_constraints_get()
```
later we can replace uses of `pm_policy_state_lock_put` since its the same but with a "list" of size 1